karapace/protobuf/compare_type_storage.py

```diff
@@ -0,0 +1,130 @@
+from karapace.protobuf.compare_result import CompareResult
+from karapace.protobuf.exception import IllegalArgumentException
+from karapace.protobuf.proto_type import ProtoType
+from karapace.protobuf.type_element import TypeElement
+from typing import Dict, List, Optional, Union
+
+
+def compute_name(t: ProtoType, result_path: List[str], package_name: str, types: dict) -> Optional[str]:
+    string = t.string
+
+    if string.startswith('.'):
+        name = string[1:]
+        if types.get(name):
+            return name
+        return None
+    canonical_name = list(result_path)
+    if package_name:
+        canonical_name.insert(0, package_name)
+    while len(canonical_name) > 0:
+        pretender: str = ".".join(canonical_name) + '.' + string
+        pt = types.get(pretender)
+        if pt is not None:
+            return pretender
+        canonical_name.pop()
+    if types.get(string):
+        return string
+    return None
+
+
+class CompareTypes:
+    def __init__(self, self_package_name: str, other_package_name: str, result: CompareResult):
```

Comment:
just out of curiosity, so are you going to add return type to all functions?

Reply:
yes, it is required by  PEP 484. but annotation of some functions will be not exact (instead of real class name word 'object' will be used.
in few hours I will commit it there